### PR TITLE
Add multi-line flag to semantics

### LINKED ISSUE #34265: [web][upstream] In semantics upstream isMultiline
Upstream isMultiline in semantics.dart

https://github.com/flutter/flutter_web/blob/master/packages/flutter_web/lib/src/semantics/semantics.dart

### PR DESCRIPTION
**Pull request by mdebbar**
## Description

This flag is currently only used in the web engine to decide whether to create an `<input>` or `<textarea>` element.

This PR depends on the engine PR: https://github.com/flutter/engine/pull/9850

## Related Issues

Fixes https://github.com/flutter/flutter/issues/34265

## Tests

I added the following tests:

* `test/widgets/editable_text_test.dart`: "EditableText sets multi-line flag in semantics"

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
You'll also need to update: https://github.com/flutter/flutter/blob/master/packages/flutter_test/lib/src/matchers.dart#L397

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -3673,6 +3675,17 @@ class RenderSemanticsAnnotations extends RenderProxyBox {
     markNeedsSemanticsUpdate();
   }
 
+  /// If non-null, sets the [SemanticsNode.isMultiline] semantic to the given
+  /// value.
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: Same as other text-related fields. Not all semantics nodes are about text editing.

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 2]
```diff
@@ -910,6 +910,67 @@ void main() {
...
+              maxLines: 1,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(
+      semantics,
+      includesNodeWith(flags: <SemanticsFlag>[SemanticsFlag.isTextField]),
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply packages/flutter/test/widgets/editable_text_test.dart]: `RenderEditable` doesn't set the `isEnabled` flag.

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**